src/Symfony/EventListener/DeserializeListener.php

```diff
@@ -96,9 +96,31 @@ public function onKernelRequest(RequestEvent $event): void
             $context[AbstractNormalizer::OBJECT_TO_POPULATE] = $data;
         }
         try {
+            $inputData = $this->serializer->deserialize(
+                $request->getContent(),
+                $context['resource_class'],
+                $format,
+                $context
+            );
+
+            /**
+             * If deserialized request object is an instance of an Input DTO overwrite `previous_data`
+             * with `data` in order to be able to use `previous_data` for updates in State Processor
+             * without the need to load the original object explicitly.
+             */
+            foreach ($operation->getInput() as $input) {
```

Comment:
Not possible, it's an array because of legacy usage which was `['class' => Foo::class]` but this looks wrong. Also this probably breaks lots of tests